apps/cms/src/customize/graphql/mutation/inviteTest.ts

```diff
@@ -0,0 +1,210 @@
+import { emailTemplate } from '../../../template/email-invite-candidate';
+import _ from 'lodash';
+
+type IndividualTest = {
+  email: string;
+  linkTestInvite: string;
+};
+
+type Candidate = {
+  firstName?: string;
+  email: string;
+  publishedAt: string;
+};
+
+const { ApplicationError } = utils.errors;
+
+const validateEmails = (emails: string[]) => {
+  if (!emails.every((e: string) => isEmail(e))) {
+    return true;
+  }
+
+  return;
+};
+
+const validateTestId = async (testId: string) => {
+  const testTemplate = await strapi.db
+    .query('api::test.test')
+    .findOne({ where: { id: testId } });
+
+  if (!testTemplate) {
+    return true;
+  }
+  return;
+};
+
+const getNonExistingCandidates = async (emails: string[]) => {
+  try {
+    const nonExistCandidates: Candidate[] = [];
+    await Promise.all(
+      emails.map(async (email: string) => {
+        const candidate = await strapi.db
+          .query('api::candidate.candidate')
+          .findOne({ where: { email } });
+
+        if (!candidate) {
+          nonExistCandidates.push({
+            firstName: email.split('@')[0],
+            email,
+            publishedAt: new Date().toISOString(),
+          });
+        }
+      })
+    );
+
+    return nonExistCandidates;
+  } catch (error) {
+    throw new ApplicationError(error.message);
+  }
+};
+
+const createManyCandidate = async (listCandidateToCreate: Candidate[]) => {
+  try {
+    if (listCandidateToCreate.length > 0) {
+      await strapi.db.query('api::candidate.candidate').createMany({
+        data: listCandidateToCreate,
+      });
+    }
+  } catch (error) {
+    throw new ApplicationError(error.message);
+  }
+};
+
+const createIndividualTests = async (emails: string[], testId: string) => {
+  try {
+    await Promise.all(
+      emails.map(async (email: string) => {
+        // If current candidate already invited to inviting test - do nothing
+        const isIndTestExist = await strapi.db
+          .query('api::individual-test.individual-test')
+          .findOne({
+            where: {
+              $and: [{ candidate: { email }, test: { id: testId } }],
+            },
+          });
+
+        if (!isIndTestExist) {
+          const currentCandidate = await strapi.db
+            .query('api::candidate.candidate')
+            .findOne({ where: { email } });
+
+          // todo currently createMany doesn't work correctly, find a way to use
+          await strapi.db.query('api::individual-test.individual-test').create({
+            data: {
+              test: testId,
+              candidate: currentCandidate.id,
+              publishedAt: new Date().toISOString(),
+            },
+          });
+        }
+      })
+    );
+  } catch (error) {
+    throw new ApplicationError(error.message);
+  }
+};
+
+const getIndTests = async (emails: string[], testId: string) => {
+  try {
+    const listOfIndTest = await Promise.all(
+      emails.map(async (email: string) => {
```

Comment:
To reduce queries to the database

`individualTests` = await strapi.db.query('api::individual-test.individual-test').findMany({ where candidate in candidateIds and id = testId })